### ISSUE TITLE
CSS JSON file references not escaped during kickstart

### ISSUE THREAD

**Issue by atrauzzi** [FusionAuth/fusionauth-issues#565]
## CSS JSON file references not escaped during kickstart

When I reference a file in a kickstart (`@{}` syntax), the file has to have some prep already done on it so that it can be provided as a JSON string.  

### Expected behavior
Any files that I reference from a kickstart should be encoded as needed prior to being sent to the server without any explicit effort on my end to perform that encoding.

### Platform
Docker

### Additional context
With the current behaviour, if I'm committing my kickstart to source control, I either have an extra step that I have to perform to master the file prior to the kickstart, or I have to store the mastered file in source control and always remember to re-master it when making changes.

**Comment by robotdan** [comment]
The design point it to allow text files to be included and pulled into the JSON, so we do some JSON escaping to support this. 

Do you have a specific use case that is not working?

**Comment by atrauzzi** [comment]
Yeah, basically anything I try to include right now is causing kickstart to throw exceptions.

**Comment by robotdan** [comment]
To clarify, 

> All JSON file references not escaped during kickstart
> When I reference a file in a kickstart (@{} syntax), the file has to have some prep already done on it so that it can be provided as a JSON string.

`@{}` is a text include, not a JSON include, no escaping is performed. This is expected and working as designed. This syntax will simply copy and paste the contents of that file. 

https://fusionauth.io/docs/v1/tech/installation-guide/kickstart#include-text-files

`&{}` is a JSON include syntax, JSON escaping is performed. 

https://fusionauth.io/docs/v1/tech/installation-guide/kickstart#include-json-files

If you have a string that requires JSON escaping that you cannot use with the `&{}` syntax then please provide a short example in the description so we can perform a recreate and document the issue. 

It may be there is a use case for a third type of include, or perhaps a use case that is not currently covered by the `@{}` include option. 

Thanks!

**Comment by trex** [comment]
@atrauzzi I believe we tracked down the issue.  The theme API request defined in your kickstart file is is incomplete, and requires the rest of the templates to be present.  

If you are providing any templates to the theme API you must provide all of the templates. This isn't stated in the API documentation.  It looks like the validation error is getting swallowed by the kickstart logger, which isn't helpful. 

I will update the API documentation and look at improving the logging in kickstart.

**Comment by atrauzzi** [comment]
Would it be possible to update the API to not require everything to be specified?

**Comment by atrauzzi** [comment]
> The thinking is that if you are building a theme, you likely will want to theme all pages.

If someone only needed to customize the CSS and a handful of templates, why force them to boilerplate the rest in?

**Comment by robotdan** [comment]
Here is the reason, if you change the helpers file - this contains all of the macros used in every other template. 

If you modify the helpers file, we have no way to know if every other template is broken or not. It is also likely you don't know what buttons or paths your users will take through the site. This means that if you theme some pages and not others your users will see a mix of styles. 

This is designed to encourage / enforce the users of the theme to theme all pages. It may not be ideal for all use cases. 

If you all you want to do is change the CSS - this is supported. Once you modify one template, all of them are required to be provided. 

The alternative is to use a Copy (create with sourceThemeId) and then do a PATCH to update just a few templates.

**Comment by atrauzzi** [comment]
Gotcha!  That all makes sense.  We'll make sure to strive to customize the system holistically :smile: 

BTW, now that this is working, it's a dream.  Thank you!
